crates/ark/src/lsp/completions/function_context.rs

```diff
@@ -0,0 +1,228 @@
+//
+// function_context.rs
+//
+// Copyright (C) 2025 Posit Software, PBC. All rights reserved.
+//
+//
+
+use tower_lsp::lsp_types::Range;
+use tree_sitter::Node;
+
+use crate::lsp::document_context::DocumentContext;
+use crate::lsp::encoding::convert_tree_sitter_range_to_lsp_range;
+use crate::treesitter::node_find_parent_call;
+use crate::treesitter::node_text;
+use crate::treesitter::BinaryOperatorType;
+use crate::treesitter::NodeType;
+use crate::treesitter::NodeTypeExt;
+use crate::treesitter::UnaryOperatorType;
+
+/// Represents how a function is being used in an expression
+#[derive(Clone, Debug, PartialEq)]
+pub(crate) enum FunctionUsage {
+    /// Function is being called, e.g., `foo()`
+    Call,
+    /// Function is being referenced without calling, e.g., `foo` in `debug(foo)`
+    Reference,
+}
+
+#[derive(Clone, Debug)]
+pub(crate) struct FunctionContext {
```

Comment:
I wonder if there should be a `namespace: Option<String>` field?

Reply:
No worries but just to be clear, I thought about this to make it clear from the type definition that this struct handles reference paths, i.e. for `foo::bar` you can expect `name` to be `bar` if you see that there is an optional namespace field.